From 7780cdbfdfd3d226e4707639ad89f4b1f0b5bb60 Mon Sep 17 00:00:00 2001
From: Sven Nierlein <sven@nierlein.de>
Date: Tue, 4 Apr 2017 14:27:54 +0200
Subject: [PATCH] fix segfault in notifications when there is no eventbroker

nebcallback resultset may be null if eventbroker options prevent
notifications from beeing brokered. Do not segfault in those
cases.

Signed-off-by: Sven Nierlein <sven@nierlein.de>
---
 src/naemon/nebmods.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/naemon/nebmods.c b/src/naemon/nebmods.c
index ba6d39f..3f9dbc5 100644
--- a/src/naemon/nebmods.c
+++ b/src/naemon/nebmods.c
@@ -369,6 +369,8 @@ static neb_cb_resultset *neb_cb_resultset_create(void)
 
 void neb_cb_resultset_destroy(neb_cb_resultset *cbrs)
 {
+	if(cbrs == NULL)
+		return;
 	g_ptr_array_free(cbrs->cb_results, TRUE);
 	nm_free(cbrs);
 }
-- 
2.1.4

